### DISCUSSION TITLE
eslint error when I don't pass mutation to useEffect dependency

### DISCUSSION THREAD

**Discussion by arashi-dev** [TanStack/query#3566] [Q&A] [answered]
Hello.
my code is something like this:
```ts
const createProduct = useMutation(...)

useEffect(() => {
    createProduct.mutateAsync()
}, [])
```

ESlint gives me warning that I should specify `createProduct` in dependency array.
but if I do, it will make an infinite loop. how can I solve that?

**Edit:**
honestly, I'm not using `useMutation` directly but I have a helper custom hook which returns the the value returned by `useMutation` (createProduct). just wanted to mention this

**Comment by TkDodo** [comment, marked as answer]
the object returned from `useMutation` is not referentially stable, but `mutateAsync` itself is:

```
const { mutateAsync } = useMutation(...)

useEffect(() => {
   mutateAsync()
}, [mutateAsync])
```

**Reply by arashi-dev** [reply]
great. thank you

**Reply by Zerebokep** [reply]
@TkDodo is `mutate` also referentially stable? (instead using `mutateAsync`)

**Reply by TkDodo** [reply]
yes, it is.
